src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessWaitState.Unix.cs

```diff
@@ -200,7 +200,7 @@ internal void ReleaseRef()
         /// <summary>Associated process is a child that can use the terminal.</summary>
         private readonly bool _usesTerminal;
 
-        /// <summary>If a wait operation is in progress, the Task that represents it; otherwise, null.</summary>
+        /// <summary>If a wait operation is in progress or has completed, the Task that represents it; otherwise, null.</summary>
```

Comment:
It seems this gets only set to a task that waits until the exit competes fully (i.e. not cancelled).
Maybe name this `_waitForExit`?

Reply:
> It seems this gets only set to a task that waits until the exit competes fully (i.e. not cancelled).

That's not the case.  If the cancellation token supplied to WaitForExitAsync has cancellation requested, the task can exit out of its loop early even though the process hasn't yet exited.

Reply:
I was about to say here:
https://github.com/dotnet/runtime/blob/fd0be62f0b243378ba626dea63a562cd3e8abaa0/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessWaitState.Unix.cs#L462-L466
and now see that it's missing the assignment to _waitInProgress, which it needs.